yrs/src/block.rs

```diff
@@ -573,16 +546,39 @@ impl Item {
         } else {
             0
         };
+        let len = content.len(OffsetKind::Utf16);
         Item {
             id,
+            len,
             left,
             right,
             origin,
             right_origin,
             content,
             parent,
             parent_sub,
-            info: info,
+            info,
+        }
+    }
+
+    pub fn try_eq(&self, other: &Item) -> Option<bool> {
```

Comment:
I changed `Item::eq` in a way that assumes, that two items may have different neighbor `BlockPtr`'s, which are not structurally equatable, but still may point to the same blocks. Problem is that establishing equality in such way requires access to block store (in order to navigate to left/right neighbor), therefore without having it it's not always possible to do so.